src/rules/declaration-empty-line-before/index.js

```diff
@@ -0,0 +1,101 @@
+import {
+  blockString,
+  isCustomProperty,
+  isSingleLineString,
+  isStandardSyntaxDeclaration,
+  isStandardSyntaxProperty,
+  optionsHaveException,
+  optionsHaveIgnored,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "declaration-empty-line-before"
+
+export const messages = ruleMessages(ruleName, {
+  expected: "Expected empty line before declaration",
+  rejected: "Unexpected empty line before declaration",
+})
+
+export default function (expectation, options) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        "always",
+        "never",
+      ],
+    }, {
+      actual: options,
+      possible: {
+        except: [
+          "first-nested",
+          "after-comment",
+          "after-declaration",
+        ],
+        ignore: [
+          "inside-single-line-block",
+        ],
+      },
+      optional: true,
+    })
+    if (!validOptions) { return }
+
+    root.walkDecls(decl => {
+      const { prop, parent } = decl
+
+      if (!isStandardSyntaxDeclaration(decl)) { return }
+      if (!isStandardSyntaxProperty(prop)) { return }
```

Comment:
I'm not sure I agree here. Maybe I'm misunderstanding, but it looks to me like we're currently adding code to deliberately ignore non-standard syntax that is not problematic. Given the infinite possibilities for non-standard syntax, I don't think we can reliably hold to a principle like "Consistently ignore non-standard syntax". I think the best we can do is a more pragmatic one, like "Ignore problematic non-standard syntax". By that principle, we wouldn't bother *adding code* to ignore non-standard syntax that we don't anticipate causing problems. And we may make a little accommodation that addresses non-standard syntax because we see that one line of code will reduce fuss and improve UX.